### PR TITLE
[Merged by Bors] - feat(Trigonometric): add lemmas about `cos x = -1 ↔ _` etc

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]
Looks straightforward; I don't know the deeper library implications.
Might be worth waiting for the zulip discussion (about the order of 2, k and pi) to settle down.

**Comment by urkud** [comment]
IMHO, we should merge this without waiting for a decision about the new normal form. It's possible that the new normal form will be something like `a ≡ π [PMOD 2 * π]`, thus require rewriting of all these lemmas, old and new.

**Review by Parcly-Taxel — APPROVED** [review]

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean [hunk 1]
```diff
@@ -98,6 +98,22 @@ theorem sin_eq_sin_iff {x y : ℂ} :
   refine' exists_congr fun k => or_congr _ _ <;> refine' Eq.congr rfl _ <;> field_simp <;> ring
 #align complex.sin_eq_sin_iff Complex.sin_eq_sin_iff
 
+theorem cos_eq_one_iff {x : ℂ} : cos x = 1 ↔ ∃ k : ℤ, k * (2 * π) = x := by
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:101]: Why don't you write it as
```suggestion
theorem cos_eq_one_iff {x : ℂ} : cos x = 1 ↔ ∃ k : ℤ, x = k * (2 * π) := by
```
with the `x` on the left, like in the other statements of the file? Same thing below.

Comment (urkud) [reply Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:101]: It aligns with [Real.cos_eq_one_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/Trigonometric/Basic.html#Real.cos_eq_one_iff). Probably, we should have a convention about this. I'll ask on Zulip.

Comment (YaelDillies) [reply Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:101]: I agree with the current spelling that Yury uses, but an even better one would use `AddCommGroup.ModEq`

**Pull request closed without merging**